AK/StdLibExtras.h

```diff
@@ -90,16 +90,28 @@ constexpr SizeType array_size(T (&)[N])
     return N;
 }
 
-template<typename T>
-constexpr T min(T const& a, IdentityType<T> const& b)
+constexpr auto min(auto&& a, auto&& b)
+requires IsSame<RemoveCVReference<decltype(a)>, RemoveCVReference<decltype(b)>>
+{
+    return b < a ? forward<decltype(b)>(b) : forward<decltype(a)>(a);
+}
+
+constexpr auto min(auto& a, auto& b) -> auto&
```

Comment:
While this allows things like `min(a, b) = 3`, it's one more step away from C++'s standard.
Maybe we should keep `const&` for now.

Reply:
the more pressing point is avoiding dangling refs which the standard suffers.